pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -533,7 +534,8 @@ In JUnit 4, only methods annotated with the @Test annotation are executed.
             <property name="xpath">
                 <value>
 <![CDATA[
-//ClassOrInterfaceBodyDeclaration[MethodDeclaration[@Public='true']/MethodDeclarator[starts-with(@Image,'test')]]
+//ClassOrInterfaceDeclaration[ends-with(@Image, 'Test') or ExtendsList/ClassOrInterfaceType[typeof(@Image, 'junit.framework.TestCase', 'TestCase')]]
+    /ClassOrInterfaceBody/ClassOrInterfaceBodyDeclaration[MethodDeclaration[@Public='true']/MethodDeclarator[starts-with(@Image,'test')]]
 [count(Annotation//Name[@Image='Test'])=0]
```

Comment:
the count is not needed here, we just check for existance... and we should use `typeof` for the annotation type just in case